syntax = "proto3";

service DroneControl {
    rpc DroneUpdate (DroneUpdateCommand) returns (DroneUpdateResponse);
    rpc GetDroneInfo (GetAllDroneRequest) returns (DroneInfoReqResponse);
    rpc DroneState (DroneStateReq) returns (DroneStateResponse);
}

message DroneInfoReqResponse {
    repeated Node nodes = 1;
}

message GetAllDroneRequest {

}

message DroneStateReq{
    string id = 1;
}

message DroneStateResponse {
    Node node = 1;
}

message Node {
    string id = 1;
    float posX = 2;
    float posY = 3;
    float posZ = 4;
    float r = 5;
    float g = 6;
    float b = 7;
    float a = 8;
}

message DroneUpdateCommand {
    string id = 1;
    float posx = 2;
    float posy = 3;
    float posz = 4;
    float r = 5;
    float g = 6;
    float b = 7;
    float a = 8;
}

message DroneUpdateResponse{
    bool success = 1;
}
